### ISSUE TITLE
[FR] Scan all commits in pull request

### ISSUE THREAD

**Issue by shenxianpeng** [commit-check/commit-check-action#94]
When it is enabled, scan all commits when merging Pull Request

**Comment by shenxianpeng** [comment]
✅ **This feature has been implemented.** Starting from [v2.6.0](https://github.com/commit-check/commit-check-action/releases/tag/v2.6.0), the action now scans **all commits** in a Pull Request — not just the merge commit.

---

## How it works

When the workflow is triggered by a `pull_request` event and `message: true` is set, the action:

1. Retrieves **every commit** in the PR branch (via `git log HEAD^1..HEAD^2` or `git log origin/<base>..HEAD`)
2. Iterates over each commit message and validates it individually using `commit-check --message`
3. If **any** commit in the PR fails the check, the overall result is a failure

This means every commit in your PR must comply with the configured rules (e.g. Conventional Commits).

## What you need to do

**Nothing extra** — the feature is automatic. Just make sure your workflow includes:

```yaml
on:
  pull_request:

jobs:
  commit-check:
    steps:
      - uses: actions/checkout@v5
        with:
          fetch-depth: 0
      - uses: commit-check/commit-check-action@v2
        with:
          message: true    # 👈 this enables per-commit validation
          # other inputs...
```

That's it. The action will validate every commit in the PR against your [`commit-check.toml`](https://commit-check.github.io/commit-check/configuration.html) rules.

## Example

If a PR has 3 commits:
- `feat: add login page` ✅ passes
- `fix wrong stuff` ❌ fails (missing type prefix)
- `docs: update readme` ✅ passes

The action will report a **failure** because the second commit doesn't follow Conventional Commits, and you'll see which specific commit(s) failed in the output.

**Comment by shenxianpeng** [comment]
Closed as completed — the feature to scan all commits in a pull request has been implemented since v2.6.0. See the comment above for details.
